ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -459,26 +459,34 @@ applyPatchGhcPrim ghcFlavor = do
-              "\n#if MIN_VERSION_ghc_prim(0, 7, 0)\n           then return (HValue (unsafeCoerce linked_bco))\n           else case mkApUpd0# linked_bco of { (# final_bco #) ->\n#else\n           then return (HValue (unsafeCoerce# bco#))\n           else case mkApUpd0# bco# of { (# final_bco #) ->\n#endif" .
-          replace
-              "bco <- linkBCO' arr bco\n      writePtrsArrayBCO i bco marr"
-              "\n#if MIN_VERSION_ghc_prim(0, 7, 0)\n      bco <- linkBCO' arr bco\n      writePtrsArrayBCO i bco marr\n#else\n      BCO bco# <- linkBCO' arr bco\n      writePtrsArrayBCO i bco# marr\n#endif" .
-          replace
-              "writePtrsArrayBCO :: Int -> BCO -> PtrsArr -> IO ()"
-              "\n#if MIN_VERSION_ghc_prim(0, 7, 0)\nwritePtrsArrayBCO :: Int -> BCO -> PtrsArr -> IO ()\n#else\nwritePtrsArrayBCO :: Int -> BCO# -> PtrsArr -> IO ()\n#endif" .
-          replace
-              "writePtrsArrayMBA :: Int -> MutableByteArray# s -> PtrsArr -> IO ()"
-              "\n#if MIN_VERSION_ghc_prim(0, 7, 0)\nwritePtrsArrayMBA :: Int -> MutableByteArray# s -> PtrsArr -> IO ()\n#else\ndata BCO = BCO BCO#\nwritePtrsArrayMBA :: Int -> MutableByteArray# s -> PtrsArr -> IO ()\n#endif" .
-          replace
-              "newBCO# instrs lits ptrs arity bitmap s"
-              "\n#if MIN_VERSION_ghc_prim(0, 7, 0)\n  newBCO# instrs lits ptrs arity bitmap s\n#else\n  case newBCO# instrs lits ptrs arity bitmap s of\n    (# s1, bco #) -> (# s1, BCO bco #)\n#endif"
+              "do BCO bco# <- linkBCO' arr bco" .
+          replaceIfGhcPrim070Else 11
+            "then return (HValue (unsafeCoerce linked_bco))\n           else case mkApUpd0# linked_bco of { (# final_bco #) ->"
+            "then return (HValue (unsafeCoerce# bco#))\n           else case mkApUpd0# bco# of { (# final_bco #) ->" .
+          replaceIfGhcPrim070Else 6
+            "bco <- linkBCO' arr bco\n      writePtrsArrayBCO i bco marr"
+            "BCO bco# <- linkBCO' arr bco\n      writePtrsArrayBCO i bco# marr" .
+          replaceIfGhcPrim070Else 0
+            "writePtrsArrayBCO :: Int -> BCO -> PtrsArr -> IO ()"
+            "writePtrsArrayBCO :: Int -> BCO# -> PtrsArr -> IO ()" .
+          replaceIfGhcPrim070Else  0
+            "writePtrsArrayMBA :: Int -> MutableByteArray# s -> PtrsArr -> IO ()"
+            "data BCO = BCO BCO#\nwritePtrsArrayMBA :: Int -> MutableByteArray# s -> PtrsArr -> IO ()" .
+          replaceIfGhcPrim070Else 2
+            "newBCO# instrs lits ptrs arity bitmap s"
+            "case newBCO# instrs lits ptrs arity bitmap s of\n    (# s1, bco #) -> (# s1, BCO bco #)"
         =<< readFile' createBCO
         )
+  where
+    replaceIfGhcPrim070Else :: Int -> String -> String -> String -> String
+    replaceIfGhcPrim070Else n s r = replace s (ifGhcPrim070Else n s r)
+
+    ifGhcPrim070Else :: Int -> String -> String -> String
+    ifGhcPrim070Else n s r =
```

Comment:
Relying on the caller passing in the indentation seems a bit fragile. We could instead see how much indentation there already is and just use that.